network/p2p/peerstore/peerstore.go

```diff
@@ -97,7 +97,7 @@ func MakePhonebook(connectionsRateLimitingCount uint,
 }
 
 // GetAddresses returns up to N addresses, but may return fewer
-func (ps *PeerStore) GetAddresses(n int, role phonebook.PhoneBookEntryRoles) []*peer.AddrInfo {
+func (ps *PeerStore) GetAddresses(n int, role phonebook.Roles) []*peer.AddrInfo {
```

Comment:
The address match the roles exactly, or _any_? Unclear now since `role` has become a set.

Reply:
refactored to have  a single role `role phonebook.Role`